### PR TITLE
chore(broker-core): transform only records which are read by an exporter

### LINKED ISSUE #2458: Spike: transform only records which are read by an exporter
**Motivation**
* record transformation for the exporter can be expensive - especially for large/many variables

**Goal**
* reduce the record transformation whenever it is possible (e.g. the record type is not processed by the exporter)

**Ideas**
* skip records which are not processed
* do the transformation in a lazy way (i.e. only when it is processed)

### PR DESCRIPTION
**Pull request by saig0**
* transform record value lazy
* transform record's variables lazy
* every exporter can provide a record filter
* build an event filter by aggregation all exporter record filters

closes #2458

### CODE REVIEW

**Comment by saig0** [comment]
@Zelldon, it would be good if you could have a look at this today :)

**Review by ChrisKujawa — APPROVED** [review]
Nice thanks :+1:

**Comment by Zelldon** [comment]
Could you please fix your pr before leaving :see_no_evil:

**Comment by saig0** [comment]
bors retry

File: exporters/elasticsearch-exporter/src/main/java/io/zeebe/exporter/ElasticsearchExporter.java [hunk 1]
```diff
@@ -76,10 +80,6 @@ public void export(Record record) {
       createIndexTemplates();
     }
 
-    if (configuration.shouldIndexRecord(record)) {
-      client.index(record);
```
Comment (Zelldon) [inline exporters/elasticsearch-exporter/src/main/java/io/zeebe/exporter/ElasticsearchExporter.java]: Ups :laughing:

**Pull request merged**